Use an instance of Fuseki instead of RDFLib to service queries

shawa opened the pull request:
Opening for CI

The `dinto` object was SLOW to start and really just SAD and is now gone. `queries` is now `dinto` and the functions that returned `preparedQuery` objects now just go and run the thing.

The only API changes this introduces is that `ping` now also checks that Fuseki is available, and returns a 503 if it can't connect and run a dummy query.

The address and port are configureable with `ASCLEPIUS_FUSEKI_ADDRESS`.

First-time requests all now seem to be < 1 second.

c-brenn committed:
initial Dockerfile for fuseki :older-woman:

shawa committed:
Good riddance to bad performance

shawa committed:
the queries actually do the thing

c-brenn committed:
index the rdf graph at docker build time for speedy boot

something funky was happening with `tdbloader2`'s default SORT_ARGS on alpine
causing the build to fail, so I override them with `-m` which does nothing.

shawa committed:
Now using a SPARQL endpoint for much more delightful speed

shawa committed:
fix logging not being shown, add env var for the fuseki location

shawa committed:
have /ping check that Asclepius can talk to its oracle

shawa committed:
make sure fuseki actually responds to queries

shawa committed:
Update README doc to reflect new ping behavior

shawa committed:
No longer need utils

shawa committed:
remove redundant tests

houli commented on asclepius/asclepius/main.py:
```diff
@@ -32,20 +34,17 @@ def handle_invalid_usage(error):
     """Return a list of all drugs listed in DINTO"""
-    return jsonify(to_labelled(dinto.all_drugs()))
+    resp = dinto.all_drugs()
+    return jsonify(dinto.all_drugs())
```
Calling `dinto.all_drugs()` twice here

c-brenn replied:
removed duplicate call

houli commented on asclepius/asclepius/main.py:
```diff
@@ -69,12 +68,23 @@ def ddis():
+                      "results": {"bindings": []}}
+    try:
+        resp = requests.post(dinto.SPARQL_ENDPOINT, data={'query': DUMMY_REQUEST})
+    except requests.exceptions.ConnectionError:
```
`RequestException` is the superclass of all the exceptions that the requests library will throw so probably better to catch that

c-brenn replied:
do we want to keep the `ping` code around seeing as there's no long boot time anymore?

22a replied:
I vote to remove all status checking

houli replied:
We can get rid of it I think

c-brenn replied:
got rid of it all. much less cruft now

shawa replied:
No need for it now, yeah. Let's dump it.

houli reviewed (changes requested):
Need to update your requirements.txt for the tests to work on CI

c-brenn committed:
tidy chiron's Dockerfile

move RUN/ENV commands around so layers are cached more easily - updating
dinto version does not require full rebuild.

c-brenn committed:
config changes so everything works together

c-brenn committed:
adds chiron to dev compose file

removes DINTO from ascelpius and adds chiron

TODO: add chiron to main compose file before iteration 0.3 tag

c-brenn commented:
updated the requirements.txt

c-brenn committed:
removes ping endpoint & duplicate call to all_drugs

no need to ping anymore because the dinto service boots quickly.

c-brenn committed:
removes ping logic from Panacea

Asclepius no longer has to wait to load DINTO, so there is no need to ping it to
see if it is ready.

c-brenn commented:
I think this is ready to go

houli commented:
Grand, will change the ASCLEPIUS_ADDRESS stuff to use the config thing in my PR when it's in
